### ISSUE TITLE
Raw ordering broken in v1.17.3

### ISSUE THREAD

**Issue by matthew-inamdar** [spatie/laravel-query-builder#266]
### The breaking change

The following line breaks the ability to have raw ordering as part of the query:

https://github.com/spatie/laravel-query-builder/blob/59d623c8f92d02f40fcd896f398103ebd4de24e6/src/Concerns/SortsQuery.php#L165

### Use case example

This is the specific use case which was working prior to `v1.17.3`:

```php
$baseQuery = Service::query()
    ->orderByRaw(
        "(
            SELECT COUNT(*) 
            FROM `service_taxonomies` 
            WHERE `service_taxonomies`.`taxonomy_id` IN ('?')
        ) DESC",
        $taxonomyIds
    )
    ->orderBy('services.name');

$services = QueryBuilder::for($baseQuery)->paginate();
```

### Explanation

This is because it expects each item in the `$query->order` array to have the `column` index. However, as you can see below, there is no `order` index for a raw order.

```php
// var_dump($query->order);

array:2 [
  0 => array:2 [
    "type" => "Raw"
    "sql" => "(SELECT COUNT(*) FROM `service_taxonomies` WHERE `service_taxonomies`.`taxonomy_id` IN ('?')) DESC"
  ]
  1 => array:2 [
    "column" => "services.name"
    "direction" => "asc"
  ]
]
```

### Versions effected

Confirmed that this functionality is working in `v1.17.2` and broken in `v1.17.3`

**Comment by matthew-inamdar** [comment]
This relates to #258

**Comment by AlexVanderbist** [comment]
Hey @matthew-inamdar, thanks for the in-depth explanation. I've merged the fix PR on the other issue and the problem should be resolved in 1.17.4 (will tag tag in a couple of minutes).
